src/test/java/dk/cngroup/lentils/service/StatusServiceTest.java

```diff
@@ -164,6 +164,72 @@ public void getStatusScoreForOneLockedCyper() {
         assertEquals(0, service.getStatusScore(status.getTeam(), status.getCypher()));
     }
 
+    @Test
+    public void restOfCyphersAreLockedIsTrueTest() {
+        List<Cypher> cyphers = new LinkedList<>();
+        Cypher cypher = generator.generateNewCypher();
+        cypher.setStage(1);
+        cyphers.add(cypher);
+        Team team = generator.generateValidTeam();
+        when (cypherRepository.findByStageGreaterThanOrderByStageAsc(1)).thenReturn(cyphers);
+        when (statusRepository.existsStatusByCypherAndTeamAndCypherStatus(any(), any(), any()))
+                .thenReturn(true);
+
+        assertEquals(true, service.restOfCyphersAreLocked (team, cypher));
+    }
+
+    @Test
+    public void restOfCyphersAreLockedIsFalseTest() {
+        List<Cypher> cyphers = new LinkedList<>();
+        Cypher cypher = generator.generateNewCypher();
+        cypher.setStage(1);
+        cyphers.add(cypher);
+        Team team = generator.generateValidTeam();
+        when (cypherRepository.findByStageGreaterThanOrderByStageAsc(1)).thenReturn(cyphers);
+        when (statusRepository.existsStatusByCypherAndTeamAndCypherStatus(any(), any(), any()))
+                .thenReturn(false);
+
+        assertEquals(false, service.restOfCyphersAreLocked (team, cypher));
+    }
+
+    @Test
+    public void getAllByTeamSortByCypherStageTest() {
```

Comment:
Tenhle test jsem možná předtím přehlédl, jestli není nedávno přidaný. Moje chyba.

Ale ten bych napsal spíš jako integrační. Tzn. nějakým způsobem nahrál statusy do db a zkusil otestovat pořadí, které se vrátí.

Tady je totiž chybka v tom, že ty mockuješ návratovou hodnotu findByTeamOrderByCypherStageAsc a pak testuješ co vrátí.

Tzn. testuješ to, co nastavít tady v testu, ale ne to, co ta služba vlastně provede a vrátí. Je to takový zbytečný test, jestli to dobře chápu.

Reply:
jo, byl totiz puvodne napsany na kontrolu toho setrideni, sifry se tam zadavaly v opacnem poradi (stage od 3 do 1) a kontrolovala jsem tim, jestli se to opravdu setridi. Pote, co jsem to setrideni opravila na databazove ztratil smysl, stejne jako ten nasledujici (musela jsem tam zmenit to poradi zadavani sifer, aby fungoval) a chtela jsem se zeptat, jestli by nebylo lepsi ho smazat, ale zapomnela jsem.

Reply:
Chápu. V tom případě bych to tady smazal a když tak napsal přímo speciální test na to třídění někde kde bude pasovat, jestli ho chceš mít.